### DISCUSSION TITLE
User subroutine for beam elements

### DISCUSSION THREAD

**Discussion by Togela** [OpenRadioss/OpenRadioss#3026] [Q&A]
Hello everyone,
I want to create my own user subroutines for beam or truss elements. As far as I can see, openradioss only has an interface for solid and shell elements implemented. How difficult is it to extend openradioss for beam elements?

Regards Tobias

**Comment by MarianBulla** [comment]
Hello @Togela,

Thank you very much for your interest in OpenRadioss and its USER interface.
It is very simple creating 1d-elements (spring or truss or beam),

Please, have a look in the documentation.

There is also an example for a 1d-element:  https://help.altair.com/hwsolvers/rad/topics/solvers/rad/user_property_elements_user_spring_element_example_r.htm

Hope it helps you ?

Thanks and best regards,
Marian

**Reply by Togela** [reply]
Hello Marian,
thank you for your quick reply. 
Sorry for not making my intentions clear:  Instead of changing the element cross-section properties, I'd like to write my own material models for beam and truss elements.
